test/integration_test/ncengine/graphics2/StructuredBuffer_tests.cpp

```diff
@@ -0,0 +1,140 @@
+#include "DiligentEngineFixture.inl"
+#include "graphics2/diligent/resource/base/StructuredBuffer.h"
+#include "graphics2/diligent/resource/MaterialResourceSignature.h"
+#include "graphics2/frontend/subsystem/MaterialRegistry.h"
+#include "ncengine/graphics/Material.h"
+
+#include <array>
+
+class StructuredBufferTest : public DiligentEngineFixture
+{
+    protected:
+        static constexpr auto signatureName = "testSignature";
+        static constexpr auto variableName = "testBuffer";
+        static constexpr auto bindingIndex = uint8_t{1};
+        static constexpr auto maxInstanceCount = 10u;
+        static constexpr auto initialInstanceCount = 3u;
+
+        std::unique_ptr<nc::graphics::MaterialResourceSignature> signature;
```

Comment:
I suppose this test doesn't have to depend on `MaterialResourceSignature` or `MaterialData` anymore.